File: framework/src/main/java/wulinpeng/com/framework/base/ui/DisplayUtil.java
package wulinpeng.com.framework.base.ui;

/**
 * @author wulinpeng
 * @datetime: 18/10/4 下午11:19
 * @description:
 */
import android.content.Context;

public class DisplayUtil {

    public static int dip2px(Context context, float dpValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

    public static int px2dip(Context context, float pxValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5f);
    }
}

File: framework/src/main/java/wulinpeng/com/framework/base/ui/DisplayUtil.kt
package wulinpeng.com.framework.base.ui

import android.content.Context

/**
 *
 * @author wulinpeng
 * @datetime: 2019/4/6 8:56 PM
 * @description:
 */
object DisplayUtil {

    @JvmStatic
    fun dip2px(context: Context, dpValue: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }

    fun px2dip(context: Context, pxValue: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (pxValue / scale + 0.5f).toInt()
    }
}